src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1

```diff
@@ -336,53 +336,45 @@ class ServiceConnection: ADOSVTBase
-                        $controlResult.AddMessage("Global groups that have access to service connection.",$restrictedGroups)
-                        $controlResult.SetStateData("Global groups that have access to service connection",$restrictedGroups)
-                        $controlResult.AdditionalInfo += "Total number of global groups that have access to service connection: " + ($restrictedGroups | Measure-Object).Count;
-                    }
-                    else{
-                        $controlResult.AddMessage([VerificationResult]::Passed,"No global groups have access to service connection.");
-                    }
+                # fail the control if restricted group found on service connection
+                if($restrictedGroups)
+                {
+                    $controlResult.AddMessage("Count of global groups that have access to service connection: ", @($restrictedGroups).Count)
+                    $controlResult.AddMessage([VerificationResult]::Failed,"Do not grant global groups access to service connections. Granting elevated permissions to these groups can risk exposure of service connections to unwarranted individuals.");
+                    $controlResult.AddMessage("Global groups that have access to service connection.",$restrictedGroups)
+                    $controlResult.SetStateData("Global groups that have access to service connection",$restrictedGroups)
+                    $controlResult.AdditionalInfo += "Count of global groups that have access to service connection: " + @($restrictedGroups).Count;
                 }
-                else {
+                else{
                     $controlResult.AddMessage([VerificationResult]::Passed,"No global groups have access to service connection.");
                 }
             }
             else {
-                $controlResult.AddMessage([VerificationResult]::Manual,"List of restricted global groups for service connection is not defined in your organization policy. Please update your ControlSettings.json as per the latest AzSK.ADO PowerShell module.");
-            }
-        }
-        catch {
-            $failMsg = $_
-            $controlResult.LogException($_)
-        }
+                $controlResult.AddMessage([VerificationResult]::Passed,"No global groups have access to service connection.");
+            }                   
 
-        if(![string]::IsNullOrEmpty($failMsg))
-        {
-            $controlResult.AddMessage([VerificationResult]::Manual,"Unable to fetch service connections details. $($failMsg)Please verify from portal that you are not granting global security groups access to service connections");
+            $restrictedGroups = $null;
+            $restrictedGlobalGroupsForSerConn = $null;
         }
+        catch {            
+            $controlResult.LogException($_)
```

Comment:
Add control result error in catch block  

$controlResult.AddMessage([VerificationResult]::Error,"Unable to fetch service connections details.")